mlir/lib/Dialect/AIR/IR/AIRDialect.cpp

```diff
@@ -1171,6 +1171,40 @@ void WaitAllOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
 // Channel op
 //
 
+LogicalResult ChannelOp::verify() {
+  if (isBroadcast()) {
+    auto bundle_size = getSize();
+    auto broadcast_shape = getBroadcastShape();
+    if (bundle_size.size() != broadcast_shape.size())
+      return emitOpError("bundle size should match broadcast_shape");
+    int diffDims = 0;
+    for (int i = 0; i < (int)bundle_size.size(); i++)
+      if (dyn_cast<IntegerAttr>(bundle_size[i]).getInt() !=
+          dyn_cast<IntegerAttr>(broadcast_shape[i]).getInt())
+        diffDims++;
+    if (diffDims > 1)
+      return emitOpError("bundle sizes and broadcast_shape should only differ "
+                         "along one dimension");
```

Comment:
Thanks for adding a verifier to check for broadcast shape's legality. For this condition, I recommend modifying it to follow Numpy broadcast's rule which says if dimension n is different between input and output, then one of the two values must be 1: https://numpy.org/doc/stable/user/basics.broadcasting.html#general-broadcasting-rules